### PR TITLE
Add method to impute mutation times

### LINKED ISSUE #2758: Add method to compute mutation time from the mutation's node
[As we discussed in the tsinfer-qc project](https://github.com/tskit-dev/tsinfer-qc/issues/23), it would be helpful to have a method to fill in the `ts.mutations_time` array with the time of the mutation's node (if it is not null). @jeromekelleher proposed something like this:

```python
mutations_time = ts.mutations_time.copy()
unknown = tskit.is_unknown_time(mutations_time)
mutations_time[unknown] = ts.nodes_time[ts.mutations_node[unknown]]
```

It seems to work in my initial testing. There is already a [`Table.Collection.compute_mutation_times()`](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TableCollection.compute_mutation_times) method, however, it computes the mutation time as the midpoint of the associated child and ancestor node times ([I think this is where the method is defined](https://github.com/tskit-dev/tskit/blob/fd725734252c355ae1ab71d8dc14704f9b98e9d5/python/tests/tsutil.py#LL795C32-L795C32)). In the case of the plot we are working on, it would be preferable to use the child node time only. 

Perhaps we could add a boolean parameter to the `compute_mutations_time` method, called something like `use_child_node_only=False`, then add the functionality to the existing method?

### PR DESCRIPTION
**Pull request by duncanMR**
Fixes #2758. I checked that the documentation renders okay and I didn't see any problems. Let me know if there are any changes I need to make!

### CODE REVIEW

File: python/tskit/trees.py [hunk 1]
```diff
@@ -8882,6 +8883,37 @@ def coalescence_time_distribution(
             span_normalise=span_normalise,
         )
 
+    def impute_mutations_time(
```
[the remark this answers is not in the document]
Comment (duncanMR) [reply python/tskit/trees.py:8886]: That's a fair point; the behaviour with unknown times might be surprising to people. I will update the docstring and name as you suggest; updating the commit now

File: python/tskit/trees.py [hunk 2]
```diff
@@ -8882,6 +8883,37 @@ def coalescence_time_distribution(
...
+        self,
+        method=None,
+    ):
+        """Returns an array of mutation times determined from the times of
+        associated nodes. Not to be confused with
+        :meth:`TableCollection.compute_mutation_times`, which modifies the
+        ``time`` column of the mutations table in place.
+
```
[the remark this answers is not in the document]
Comment (duncanMR) [reply python/tskit/trees.py:8894]: Is this okay?
```python
    def mutations_time_impute_unknown(
        self,
        method=None,
    ):
        """
        Returns an array of mutation times, where any unknown times are
        imputed from the times of associated nodes. Not to be confused with
        :meth:`TableCollection.compute_mutation_times`, which modifies the
        ``time`` column of the mutations table in place.

        :param str method: The method used to impute the unknown mutation times.
            Currently only "min" is supported, which uses the time of the node
            below the mutation as the mutation time. The "min" method can also
            be specified by ``method=None`` (Default: ``None``).
        :return: An array of length equal to the number of mutations in the
            tree sequence.
        """
```

File: python/tskit/trees.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (duncanMR) [reply python/tskit/trees.py:8886]: On second thoughts, could we perhaps make it `ts.impute_unknown_mutations_time()`? I think it sounds better to have the verb first

**Review by benjeffery — APPROVED** [review]
Awesome! The CI is currently failing with https://github.com/tskit-dev/tskit/issues/2761, but once that is fixed I'll merge this.

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**